lectures/2-math-and-crypto.tex

```diff
@@ -0,0 +1,269 @@
+\end{definition}
+
+Informally, this assumption means that given $u$, it is very hard to find $\alpha$ such that $u = g^{\alpha}$. But now we can write down this formally!
+
+\subsubsection{Computational Diffie-Hellman}
+
+Another fundamental problem in cryptography is the \textbf{Computational Diffie-Hellman} problem. It states that given $g^{\alpha},g^{\beta}$ it is hard to find $g^{\alpha\beta}$. This property is frequently used in the construction of cryptographic protocols such as the Diffie-Hellman key exchange.
+
+Let us define this problem formally.
+
+\begin{definition}
+    Let $\mathbb{G}$ be a cyclic group of prime order $r$ generated by $g \in \mathbb{G}$. Define the following game:
+    \begin{enumerate}
+        \item Both challenger $\mathcal{C}h$ and adversary $\mathcal{A}$ take a description $\mathbb{G}$ as an input: order $r$ and generator $g \in \mathbb{G}$.
+        \item $\mathcal{C}h$ computes $\alpha, \beta \xleftarrow{R} \mathbb{Z}_r, u \gets g^{\alpha}, v \gets g^{\beta}, w \gets g^{\alpha\beta}$ and sends $u,v \in \mathbb{G}$ to $\mathcal{A}$.
+        \item The adversary $\mathcal{A}$ outputs $\hat{w} \in \mathbb{G}$.
+    \end{enumerate}
+
+    We define $\mathcal{A}$'s \textbf{advantage in solving the computational Diffie-Hellman problem in $\mathbb{G}$}, denoted as $\text{CDH}\mathsf{adv}[\mathcal{A},\mathbb{G}]$, as the probability that $\hat{w} = w$.
+\end{definition}
+
+\begin{definition}
+    The \textbf{Computational Diffie-Hellman Assumption} holds in the group $\mathbb{G}$ if for any efficient adversary $\mathcal{A}$ the advantage $\text{CDH}\mathsf{adv}[\mathcal{A},\mathbb{G}]$ is negligible.
+\end{definition}
+
+\subsubsection{Why this is needed?}
+
+Typically, it is impossible to prove the predicate ``for every efficient adversary $\mathcal{A}$ this probability is negligible'' and therefore we need to make assumptions, such as the Discrete Logarithm Assumption or the Computational Diffie-Hellman Assumption. In turn, proving the statement ``if $X$ is secure then $Y$ is also secure'' is manageable and does not require solving any fundamental problems. So, for example,
+knowing that the probability of the adversary to break the Diffie-Hellman assumption is negligible, we can prove that the Diffie-Hellman key exchange is secure. 
+
+\subsection{Basic Number Theory}
+
+\subsubsection{Primes}
+
+Primes are often used when doing almost any cryptographic computation. A prime number is a natural number ($\mathbb{Z}^{+}$) that is not a product
+of two smaller natural number. In other words, the prime number is divisible only by itself and 1. First primes look like this: 2, 3, 5, 7, 11...
+
+\subsubsection{Deterministic prime tests}
+
+A primality test is deterministic if it outputs $\mathsf{True}$ when the number is a prime and $\mathsf{False}$ when the input is composite with probability 1.
```

Comment:
I believe you might use `\texttt` for true and false.

Besides, I believe that the definition of determinism is a sense that the probability is strictly $1$ is not entirely correct.

Deterministic means that by calling the algorithm with two same inputs, the algorithm's inner workings would be identical in both cases. The result (true/false) might be 1, even for probabilistic algorithms.